### PR TITLE
Fix: llm completion exception breaks CodeActAgent

### PR DESCRIPTION
**Pull request by shubhamofbce**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**
This fixes the issue mentioned in #3575
Added a try catch block to catch any exception thrown by llm.completion.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
I just added a try catch block to catch any exception, Also the step method needs to return and action, So I am returning the AgentFinishAction with an error message thought to inform user that there was some error and finish action allows the user to retry it or give a new input.

---
**Link of any specific issues this addresses**
#3575 

https://github.com/user-attachments/assets/8d833e8b-aa92-40a5-bee0-9f06d7205449

### CODE REVIEW

File: agenthub/codeact_agent/codeact_agent.py [hunk 1]
```diff
@@ -198,8 +199,13 @@ def step(self, state: State) -> Action:
             params['extra_headers'] = {
                 'anthropic-beta': 'prompt-caching-2024-07-31',
             }
-
-        response = self.llm.completion(**params)
+        try:
+            response = self.llm.completion(**params)
+        except Exception as e:
+            logger.error(f'LLM completion failed: {e}')
```
Comment (tobitege) [inline agenthub/codeact_agent/codeact_agent.py:205]: This logger is not needed, I think, as in llm.py the completion method itself already logs the exception(s).

**Comment by tobitege** [comment]
Thanks for working on this!

**Comment by tobitege** [comment]
What could be nice is to adapt the existing exception handling in llm.py's completion methods to disable the lenghty tracebacks, but include the messages from cascaded exceptions (it is a lot of "clutter" in the log with references outside of OpenHands' codebase).

**Comment by shubhamofbce** [comment]
@tobitege  I've created this PR to address the exception issue in LLM completion. I've used AgentFinish for this, as it seemed appropriate since the action type will be "finish" in case of any exception, with an updated message using the "thought" field. However, I'm wondering if we should create a new action like AgentError instead. What are your thoughts on this approach?

**Comment by tobitege** [comment]
However, I'm wondering if we should create a new action like AgentError instead. What are your thoughts on this approach?

As a message type I'd prefer the name to reflect that it is related to the LLM, like we have `LLMResponseError` already (openhands/core/exceptions.py).
Maybe first look into that before creating a new one?

**Comment by shubhamofbce** [comment]
@tobitege Thanks for your comments, I will look into them.

**Comment by shubhamofbce** [comment]
@tobitege  I found that we can use LLMResponseError since any exception in the completion method is related to an LLM response. I’ve raised that exception directly from the completion method and made it return AgentFinishAction whenever it catches an LLMResponseError.

I’ve updated my PR with these changes. Let me know if you have any questions or suggestions.

File: agenthub/codeact_agent/codeact_agent.py (hunk 1 above)
Comment (shubhamofbce) [reply agenthub/codeact_agent/codeact_agent.py]: removed this.

**Review by tobitege — APPROVED** [review]
Thanks for your contribution! 👍

**Pull request merged**